src/cmap/connection.ts

```diff
@@ -468,6 +489,8 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     this.removeAllListeners(Connection.PINNED);
     this.removeAllListeners(Connection.UNPINNED);
```

Comment:
another cleanup thought - we can now use optional chaining on the callback

```typescript
if (typeof callback === 'function') {
  callback()
}
```

becomes 

```typescript
callback?.();
```

or an alternative is to ensure it's always defined at the top of the function:
```typescript
callback = callback ?? () => {};
```

Reply:
Yea and while we're at it let's use `return callback?.();` its a common pattern that makes it so we don't have to write `else` blocks, the callback should be the "end" of the function so returning at its invocation helps us keep that paradigm consistent.

Reply:
> let's use `return callback?.();` its a common pattern

Keep in mind that that pattern means that `callback()` can be called either synchronously or asynchronously, which is a bit of anti-pattern in Node.js, since callers would need to be prepared to handle either case